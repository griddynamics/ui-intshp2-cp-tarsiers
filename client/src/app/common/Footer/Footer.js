import React from 'react';

import { RouterLink, ExternalLink } from '../Links';
import logo from '../../../assets/logo.png';
import appConfig from '../../../config/appConfig';
import './Footer.scss';

const Footer = props => {
  const { toggleHFVisibility } = props;
  const visible = toggleHFVisibility.value;

  const renderLinks = list =>
    list.map(link => {
      const { id, value, href, isOrigin } = link;
      const linkType = isOrigin ? (
        <RouterLink href={href}>
          &bull;&nbsp;
          {value}
        </RouterLink>
      ) : (
        <ExternalLink href={href}>
          &bull;&nbsp;
          {value}
        </ExternalLink>
      );

      return <li key={id}>{linkType}</li>;
    });

  const { links, payments } = appConfig.footer;
  const footerLinks = <ul>{renderLinks(links)}</ul>;
  const footerPayments = <ul>{renderLinks(payments)}</ul>;

  return visible ? (
    <footer className="footer mt-120">
      <div className="container">
        <nav className="footer__block row">
          <div className="footer__copyright col-3">
            <img src={logo} alt="" />
            <span>Shopy &copy; 2019</span>
          </div>
          <div className="footer__links col-6">{footerLinks}</div>
          <div className="footer__payments col-2">
            <div className="payments-wrap">
              <p>Payment Methods</p>
              {footerPayments}
            </div>
          </div>
        </nav>
      </div>
    </footer>
  ) : null;
};

export default Footer;
