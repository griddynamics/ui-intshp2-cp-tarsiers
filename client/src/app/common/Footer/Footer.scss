@import '../../../styles/variables';
@import '../../../styles/mixin';

.footer {
  position: relative;
  padding: 60px 0 44px;
  background-color: $color-white;

  ul {
    list-style: none;
    & > li {
      font-size: 14px;
      line-height: 2;
    }
  }

  &__block {
    padding-top: 30px;
  }

  &__copyright {
    padding-bottom: 25px;

    & > span {
      padding-top: 20px;
      display: inline-block;
    }
  }

  &__payments {
    margin-left: auto;
  }

  &__links {
    & > ul {
      display: grid;
      grid-template-columns: repeat(3, auto);
      grid-template-rows: auto;
    }
  }

  div.payments-wrap {
    & > p {
      font-size: 16px;
      line-height: 1.8;
      color: $color-main;
      padding-bottom: 5px;
      white-space: nowrap;
    }

    & > ul {
      @include d-flex();

      a {
        display: block;
        width: 100%;
        height: 100%;
      }

      & > li {
        font-size: 0;
        height: 27px;
        width: 17%;
        max-width: 30px;

        &:nth-child(1) {
          @include icon-bg($icon-payment1);
        }
        &:nth-child(2) {
          @include icon-bg($icon-payment2);
        }
        &:nth-child(3) {
          @include icon-bg($icon-payment3);
        }
        &:nth-child(4) {
          @include icon-bg($icon-payment4);
        }
        &:nth-child(5) {
          @include icon-bg($icon-payment5);
        }
      }
    }
  }
  @media screen and (min-width: 476px) and (max-width: $media-tablet) {
    &__links {
      order: 1;
    }
  }
  @media screen and (max-width: $media-tablet) {
    &__block {
      padding: 0 5%;
    }
    &__payments {
      margin: 0 auto;
    }
  }
  @media screen and (max-width: $media-mobile) {
    &__copyright {
      text-align: center;
      & > img {
        margin: 0 auto;
      }
    }
    &__links {
      flex-direction: column;
      padding: 0 20px 20px;
      padding-left: 34%;

      & > ul {
        display: block;
      }
    }
    div.payments-wrap {
      width: 80%;
      margin: 0 auto;
      text-align: center;
    }
  }
}
