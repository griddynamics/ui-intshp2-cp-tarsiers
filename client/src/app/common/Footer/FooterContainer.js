import { connect } from 'react-redux';
import Footer from './Footer';

const mapStateToProps = state => ({
  toggleHFVisibility: state.toggleHFVisibility
});

export default connect(mapStateToProps)(Footer);
