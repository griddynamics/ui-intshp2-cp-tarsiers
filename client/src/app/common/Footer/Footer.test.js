import React from 'react';
import { shallow, mount } from 'enzyme';

import Footer from './Footer';

jest.mock('react-router-dom');
describe('<Footer />', () => {
  let props;

  beforeEach(() => {
    props = {
      toggleHFVisibility: { value: true }
    };
  });

  it('should match its snapshot', () => {
    const wrapper = shallow(<Footer {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to render based on props', () => {
    const wrapper = mount(<Footer {...props} />);

    expect(wrapper.prop('toggleHFVisibility').value).toBe(true);
    wrapper.setProps({ toggleHFVisibility: { value: false } });
    expect(wrapper.prop('toggleHFVisibility').value).toBe(false);
  });
});
