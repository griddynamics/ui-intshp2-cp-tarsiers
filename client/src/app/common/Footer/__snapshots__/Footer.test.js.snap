// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match its snapshot 1`] = `
<footer
  className="footer mt-120"
>
  <div
    className="container"
  >
    <nav
      className="footer__block row"
    >
      <div
        className="footer__copyright col-3"
      >
        <img
          alt=""
          src=""
        />
        <span>
          Shopy © 2019
        </span>
      </div>
      <div
        className="footer__links col-6"
      >
        <ul>
          <li
            key="14"
          >
            <ExternalLink
              href="https://www.google.com"
            >
              • 
              about us
            </ExternalLink>
          </li>
          <li
            key="15"
          >
            <RouterLink
              href="/products"
            >
              • 
              our products
            </RouterLink>
          </li>
          <li
            key="16"
          >
            <RouterLink
              href="/"
            >
              • 
              join us
            </RouterLink>
          </li>
          <li
            key="17"
          >
            <RouterLink
              href="/"
            >
              • 
              contact us
            </RouterLink>
          </li>
          <li
            key="18"
          >
            <RouterLink
              href="/"
            >
              • 
              terms and conditions
            </RouterLink>
          </li>
          <li
            key="19"
          >
            <RouterLink
              href="/"
            >
              • 
              live support
            </RouterLink>
          </li>
          <li
            key="20"
          >
            <RouterLink
              href="/"
            >
              • 
              support
            </RouterLink>
          </li>
          <li
            key="21"
          >
            <RouterLink
              href="/"
            >
              • 
              our privacy
            </RouterLink>
          </li>
        </ul>
      </div>
      <div
        className="footer__payments col-2"
      >
        <div
          className="payments-wrap"
        >
          <p>
            Payment Methods
          </p>
          <ul>
            <li
              key="22"
            >
              <ExternalLink
                href="https://www.americanexpress.com/"
              >
                • 
                amex
              </ExternalLink>
            </li>
            <li
              key="23"
            >
              <ExternalLink
                href="https://www.discover.com/"
              >
                • 
                discover
              </ExternalLink>
            </li>
            <li
              key="24"
            >
              <ExternalLink
                href="https://usa.visa.com/"
              >
                • 
                visa
              </ExternalLink>
            </li>
            <li
              key="25"
            >
              <ExternalLink
                href="https://www.mastercard.us"
              >
                • 
                mastercard
              </ExternalLink>
            </li>
            <li
              key="26"
            >
              <ExternalLink
                href="https://www.paypal.com"
              >
                • 
                credit
              </ExternalLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</footer>
`;
